### PR TITLE
SAA-542 first step towards refactoring code around event category prioritisation.

### PR DESCRIPTION
**Pull request by matt-crowson-moj**
Part 1 is really a code moving exercise with a little bit of change as well.

Part 2 will be bringing the EventCategory inline with our actual activity categories which have changed/moved on.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/PrisonRegimeService.kt [hunk 1]
```diff
@@ -75,3 +76,29 @@ class PrisonRegimeService(
...
 
 data class Priority(val priority: Int, val eventCategory: EventCategory? = null)
+
+data class EventPriorities(val priorities: Map<EventType, List<Priority>>) {
+
+  fun getOrDefault(eventType: EventType, category: EventCategory): Int = getOrDefault(eventType, category.name)
+
+  // TODO these need updating to reflect our category changes ...
+  fun getOrDefault(eventType: EventType, category: String? = null): Int =
+    priorities[eventType]?.fold(listOf<Priority>()) { acc, next ->
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/PrisonRegimeService.kt:86]: This has been pulled out of the transformers.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/PrisonRegimeService.kt [hunk 2]
```diff
@@ -75,3 +76,29 @@ class PrisonRegimeService(
...
+      if (next.eventCategory == null && acc.isEmpty()) {
+        listOf(next)
+      } else {
+        when (next.eventCategory) {
+          EventCategory.EDUCATION -> if (category?.startsWith("EDU") == true) listOf(next) else acc
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/PrisonRegimeService.kt:91]: The matching on this needs reviewing.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/model/ScheduledEvent.kt [hunk 3]
```diff
@@ -60,5 +60,5 @@ data class ScheduledEvent(
   val endTime: LocalTime?,
 
   @Schema(description = "The event priority")
-  val priority: Int?,
+  val priority: Int,
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/model/ScheduledEvent.kt:63]: I believe all events should all have a default priority??

Comment (timharrison-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/model/ScheduledEvent.kt:63]: Yes, I agree. Its either THE default for that type, or the prison's override value.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/PrisonRegimeService.kt (hunk 2 above)
Comment (timharrison-moj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/PrisonRegimeService.kt:91]: I think we currently have SAA_EDUCATION, SAA_LEISURE... etc..

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/ScheduledEventService.kt [hunk 4]
```diff
@@ -239,38 +232,37 @@ class ScheduledEventService(
...
+          schedules.transfers.prisonApiPrisonTransfersToScheduledEvents(
             prisonCode,
-            eventPriorities[EventType.EXTERNAL_TRANSFER],
+            eventPriorities.getOrDefault(EventType.EXTERNAL_TRANSFER),
           ),
-          schedules.adjudications.prisonApiOffenderAdjudicationsToScheduledEvents(
+          schedules.adjudications.prisonApiPrisonOffenderAdjudicationsToScheduledEvents(
             prisonCode,
-            eventPriorities[EventType.ADJUDICATION_HEARING],
+            eventPriorities.getOrDefault(EventType.ADJUDICATION_HEARING),
```
Comment (timharrison-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/ScheduledEventService.kt:257]: Yes - all much more concise, neater and clearer. Thanks.

**Review by timharrison-moj — APPROVED** [review]
LGTM.

**Pull request merged**